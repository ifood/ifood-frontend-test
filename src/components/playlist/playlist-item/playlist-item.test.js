import React from 'react'
import { render } from '@testing-library/react'
import { PlaylistItem } from '.'

describe('PlaylistItem', () => {
  let component

  beforeEach(() => {
    const playlist = {
      description:
        'A calm and relaxing Christmas piano soundtrack for the Holidays',
      external_urls: {
        spotify: 'https://open.spotify.com/playlist/37i9dQZF1DXbPHTEEyQ6Hv'
      },
      href: 'https://api.spotify.com/v1/playlists/37i9dQZF1DXbPHTEEyQ6Hv',
      id: '37i9dQZF1DXbPHTEEyQ6Hv',
      images: [
        {
          height: null,
          url:
            'https://i.scdn.co/image/ab67706f0000000299748689e4c7392f05545a1a',
          width: null
        }
      ],
      name: 'Christmas Peaceful Piano',
      snapshot_id:
        'MTU3NjU3OTU3OCwwMDAwMDA0OTAwMDAwMTZmMTM3NzM5MjIwMDAwMDE2ZDE1NTZjYWY1',
      type: 'playlist',
      uri: 'spotify:playlist:37i9dQZF1DXbPHTEEyQ6Hv'
    }

    component = render(<PlaylistItem playlist={playlist} />)
  })

  it('should render component correctly', () => {
    const { asFragment } = component
    expect(asFragment()).toMatchSnapshot()
  })

  it('should find playlist description text', () => {
    const { getByText } = component
    const linkElement = getByText(/A calm and relaxing Christmas/i)
    expect(linkElement).toBeInTheDocument()
  })
})
