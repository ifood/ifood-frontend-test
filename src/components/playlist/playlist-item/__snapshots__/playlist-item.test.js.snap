// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistItem should render component correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
  >
    <a
      class="makeStyles-link-2"
      href="https://open.spotify.com/playlist/37i9dQZF1DXbPHTEEyQ6Hv"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        class="MuiCardContent-root"
      >
        <img
          alt="Christmas Peaceful Piano"
          class="makeStyles-cover-3"
          src="https://i.scdn.co/image/ab67706f0000000299748689e4c7392f05545a1a"
        />
        <h3
          class="MuiTypography-root makeStyles-title-4 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
        >
          Christmas Peaceful Piano
        </h3>
        <span
          class="MuiTypography-root makeStyles-description-5 MuiTypography-caption MuiTypography-colorTextSecondary"
        >
          A calm and relaxing Christmas piano soundtrack for...
        </span>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
